pool.go

```diff
@@ -272,3 +302,173 @@ func (p *AuthProviderPool) Provision(name string) error {
 
 	return nil
 }
+
+var defaultKeyID = "0"
+
+// rsSource is what source will override the other
+var rsSource = []string{"key", "file", "dir"}
+
+// rsConfigSource is where config options will look
+var rsConfigSource = []string{"env", "config"} // this is how tokenSecret works
+
+func loadRSKeys(m *AuthProvider) error {
+	var _dir string
+	var _files, _keys = make(map[string]string), make(map[string]string)
+
+	for _, configSrc := range rsConfigSource {
+		switch configSrc {
+		case "config":
+			{
```

Comment:
@njones , this function is way too long. I think each case deserves its own function.

Reply:
Agreed! I'll break this down.